src/colorizer/IRenderCanvas.ts

```diff
@@ -3,35 +3,52 @@ import { Vector2 } from "three";
 import { ViewerStoreState } from "../state/slices";
 import { FrameLoadResult } from "./types";
 
-const canvasStateDeps = [
-  "dataset",
-  "featureKey",
-  "track",
-  "showTrackPath",
-  "colorRamp",
-  "colorRampRange",
-  "categoricalPalette",
-  "outlineColor",
-  "outlierDrawSettings",
-  "outOfRangeDrawSettings",
-  "inRangeLUT",
-  "vectorMotionDeltas",
-  "vectorVisible",
-  "vectorColor",
-  "vectorScaleFactor",
-  "backdropKey",
-  "backdropVisible",
-  "objectOpacity",
-  "backdropSaturation",
-  "backdropBrightness",
-] as const;
+export type RenderCanvasStateParams = Pick<
+  ViewerStoreState,
+  | "dataset"
+  | "featureKey"
+  | "track"
+  | "showTrackPath"
+  | "colorRamp"
+  | "colorRampRange"
+  | "categoricalPaletteRamp"
+  | "outlineColor"
+  | "outlierDrawSettings"
+  | "outOfRangeDrawSettings"
+  | "inRangeLUT"
+  | "vectorMotionDeltas"
+  | "vectorVisible"
+  | "vectorColor"
+  | "vectorScaleFactor"
+  | "backdropKey"
+  | "backdropVisible"
+  | "objectOpacity"
+  | "backdropSaturation"
+  | "backdropBrightness"
+>;
```

Comment:
Both of these do the same thing, but the one on the right feels a little more explicit to me? Kind of want @frasercl's opinion on this.